### ISSUE TITLE
Should sockets performance tests be inner loop?

### ISSUE THREAD

**Issue by ellismg** [dotnet/corefx#3542]
These tests have failed to meet the baseline a few times, but they pass on re-run.

http://dotnet-ci.cloudapp.net/job/dotnet_corefx_windows_debug_prtest/4407/testReport/junit/System.Net.Sockets.Performance.Tests/SocketPerformanceAPMTests/SocketPerformance_SingleSocketClientAsync_LocalHostServerAPM/

When the CI System runs jobs, it can run multiple jobs on the same machine at the same time, could that be causing problems or are there real problems here that we should be investigating?

It seems like the long term plan is that we move to @ianhays's [Performance] stuff.  Can we just adopt these changes for that instead of having yet another perf testing thing that also runs inner loop?

/cc @CIPop @davidsh @pgavlin @bartonjs.

**Comment by pgavlin** [comment]
I think that there's already been some discussion re: moving these tests to the outer loop and finding a better success metric.

**Comment by CIPop** [comment]
Yes, this is a duplicate of #3495.

I still think it's generally a bad idea to ever consider Perf tests as CI. (These are already marked as perf.)

**Comment by pgavlin** [comment]
I'll submit a PR today to move these into outer loop/benchmark

**Comment by pgavlin** [comment]
These tests were moved into [OuterLoop]  by #3548.

**Comment by stephentoub** [comment]
> I still think it's generally a bad idea to ever consider Perf tests as CI

Certainly if the tests weren't written to be run as part of a normal CI job then we shouldn't be doing so.

More generally, however, eventually I do believe we'll want perf legs as part of CI jobs, so that we can verify that core performance isn't regressed by a change in addition to functionality not being regressed.  I'm not sure whether this would apply to networking or not.

**Comment by bartonjs** [comment]
> eventually I do believe we'll want perf legs as part of CI jobs, so that we can verify that core performance isn't regressed

Right now one potential problem with that is that we allow multiple PR jobs to run on the same machines, in parallel.  So wall-clock based performance metrics can be impacted by process switching.

**Comment by stephentoub** [comment]
I'm less interested in wall clock measurements for such things than I am in instructions retired and GC allocations.
